src/lib/cid.js

```diff
@@ -1,18 +1,47 @@
-import CID from 'cids'
+import { CID } from 'multiformats'
+import * as dagCbor from '@ipld/dag-cbor'
+import * as dagPb from '@ipld/dag-pb'
 
-export function toCidOrNull (value) {
+/**
+ * @template {string} Prefix
+ * @param {any} value
+ * @param {import('multiformats/bases/interface').MultibaseDecoder<Prefix>} [base]
+ * @returns
+ */
+export function toCidOrNull (value, base) {
   if (!value) return null
   try {
-    return new CID(value)
+    return CID.asCID(value) || CID.parse(value, base)
   } catch (err) {
-    console.log('Failed to parse CID', value, err)
     return null
   }
 }
 
+/**
+ * This function is deprecated, use `getCodeOrNull` instead.
+ *
+ * `cid.codec` is deprecated, use integer "code" property instead
+ *
+ * @param {any} value
+ * @returns {string}
+ * @deprecated
+ */
 export function getCodecOrNull (value) {
   const cid = toCidOrNull(value)
-  return cid ? cid.codec : null
+
+  if (cid == null) return null
+  if (cid.code === dagCbor.code) {
+    return dagCbor.name
+  } else if (cid.code === dagPb.code) {
+    return dagPb.name
+  }
```

Comment:
you have `@ipld/dag-json` as a dependency, you could pretty easily add a `import * as dagJson from '@ipld/dag-json'` and then:

```suggestion
  switch (cid.code) {
    case dagCbor.code:
      return dagCbor.name
    case dagJson.code:
      return dagJson.name
    case dagPb.code:
      return dagPb.code
  }
```

I know you have #335 but dag-json seems like a simple value-add here.

An alternative is to just make a global map at the top:

```js
const codecNames = [dagCbor, dagJson, dagPb].reduce((p, c) => { p[c.code] = c.name; return p }, {})
```

Then replace this block here with:

```suggestion
  return codecNames[cid.code] || null
```

Up to you, dag-json isn't super widely used but we do see a bit of it, certainly more than plain `json` or `cbor`. `raw` is a **very** common codec and you could add it in here with the same pattern without a new import: `import * as raw from 'multiformats/codecs/raw'` (`json` can come for free too in the same way cause multiformats has it too).